### PR TITLE
Allow read-write access to configuration and download directories when installed as a Snap

### PR DESCRIPTION
**Pull request by richardneish**
We need write access to the $HOME/.config/exercism/ directory for `exercism config` and $HOME/exercism is the default download directory so we need write access there as well.

This should resolve issue #945

### CODE REVIEW

File: .goreleaser.yml
Comment (kotp) [inline .goreleaser.yml:105]: This is only the default installation directory, and should not be hard coded, i think, but should gather that from the configuration file. Otherwise we are likely going to have intermittent permission problems if the user changes the workspace.

Comment (richardneish) [reply .goreleaser.yml:105]: We can allow write access to any subdirectory of $HOME - would that be better?

Comment (kotp) [reply .goreleaser.yml]: I suspect that is better.  Perhaps the configuration file is not yet written at this point in time, to be able to get the workspace from the configuration.

**Review by ekingery — APPROVED** [review]
It is possible that the ideal solution is for a tighter restriction than $HOME. However, given the installation troubles we've seen, I think it is worthwhile to merge this as-is.

**Comment by ekingery** [comment]
Thanks for the contribution @richardneish! It should be reflected in the next release of the CLI, which should correspond with the Exercism V3 launch. The build should pass after getting merged with `main`.

**Comment by ekingery** [comment]
In order to merge, the build needs to pass. In order for the build to pass, we need to merge the latest changes from `main`. There was no base branch to merge into, so I created a separate PR for getting this merged: https://github.com/exercism/cli/pull/980

**Pull request closed without merging**
